### PR TITLE
Electrochromatic (dim-on-demand) windows

### PR DESCRIPTION
**Pull request by silicons**
## About The Pull Request

Generic electrochromatic windows
Craft a kit with 1 metal 1 glass
Use in hand to set ID
Slap that on a window
Make a button frame and controller with autolathe
Use in hand to set ID
Set that up
Click.

For mappers, set the vars to their values. Make sure to use ! as the first character of the Id so it's auto-obfuscated so stinky players can't guess it. Enjoy.

## Why It's Good For The Game

Electrochromatic dimming windows are fun.
Toggleable too.

## Changelog
:cl:
add: Electrochromatic windows have been added.
/:cl:

### CODE REVIEW

File: code/datums/components/crafting/recipes/recipes_misc.dm [hunk 1]
```diff
@@ -233,6 +233,15 @@
 	subcategory = CAT_TOOL
 	category = CAT_MISC
 
+/datum/crafting_recipe/electrochromatic_kit
+	name = "Electrochromatic Kit"
+	result = /obj/item/electronics/electrochromatic_kit
+	reqs = list(/obj/item/stack/sheet/metal = 1,
+				/obj/item/stack/cable_coil = 1)
+	time = 5
+	subcategory = CAT_MISCELLANEOUS
```
Comment (Trilbyspaceclone) [inline code/datums/components/crafting/recipes/recipes_misc.dm]: Place this in tools please

File: code/modules/research/designs/autolathe_desings/autolathe_designs_electronics.dm [hunk 2]
```diff
@@ -74,3 +74,11 @@
 	materials = list(/datum/material/glass = 20)
 	build_path = /obj/item/stock_parts/cell/emergency_light
 	category = list("initial", "Electronics")
+
+/datum/design/electrochromatic_control
+	name = "Electrochromatic Control Circuit"
+	id = "electrochromatic_control_circuit"
+	build_type = AUTOLATHE
+	materials = list(/datum/material/iron = 100, /datum/material/glass = 100)
```
Comment (Trilbyspaceclone) [inline code/modules/research/designs/autolathe_desings/autolathe_designs_electronics.dm:82]: maybe tweak this to be a bit more metal wise.

Comment (silicons) [reply code/modules/research/designs/autolathe_desings/autolathe_designs_electronics.dm:82]: y

Comment (Trilbyspaceclone) [reply code/modules/research/designs/autolathe_desings/autolathe_designs_electronics.dm:82]: It takes a full metal sheet to make, theirs no way that 1/20th of a metal sheet can cover a full glass sheet, even rods are 1/4th the mats to upgrade a single tile

Comment (lolman360) [reply code/modules/research/designs/autolathe_desings/autolathe_designs_electronics.dm:82]: spraycans are a way cheaper way of making glass not seethrough, and are technically "toggleable" via paint remover. These are only a little more expensive and take setup to make.

Comment (silicons) [reply code/modules/research/designs/autolathe_desings/autolathe_designs_electronics.dm:82]: what lolman said, shouldn't take that much to do this.

**Review by Linzolle — APPROVED** [review]

**Pull request merged**